### PR TITLE
include: zephyr: drivers: espi: Adjust terms to match eSPI specification 1.5

### PR DESCRIPTION
**Pull request by albertofloyd**
Replace master/slave by new controller/target terminology in eSPI spec 1.5.

Propagate eSPI VW change and macro name changes across eSPI drivers
Propagate eSPI VW changes to eSPI driver sample and eSPI test driver code
Add support for MEC172x modular card for samples/drivers/espi to sanity check changes

Changed scope of PR to ease review, the eSPI API clarifications moved to https://github.com/zephyrproject-rtos/zephyr/pull/69358

### CODE REVIEW

File: drivers/espi/espi_mchp_xec.c [hunk 1]
```diff
@@ -113,107 +113,107 @@ enum mchp_smvw_regs {
...
+ *  47h   | MSVW07 | res             | res          | res       | HOST_C10         |
+ *  4Ah   | MSVW08 | res             | res          | DNX_WARN  | res              |
+ *  50h   | SMVW06 | ESPI_OCB_3      | ESPI_OCB_2   | ESPI_OCB_1| ESPI_OCB_0       |
  */
 
 static const struct xec_signal vw_tbl[] = {
 	/* MSVW00 */
 	[ESPI_VWIRE_SIGNAL_SLP_S3]        = {MCHP_MSVW00, ESPI_VWIRE_SRC_ID0,
-					     ESPI_MASTER_TO_SLAVE},
+					     ESPI_CONTROLLER_TO_TARGET},
```
Comment (keith-zephyr) [inline drivers/espi/espi_mchp_xec.c:145]: This commit should be squashed with the change the header file.  Otherwise this will break bi-sectability.

**Review by keith-zephyr — COMMENTED** [review]
Happy to see this change.  Can you also update https://docs.zephyrproject.org/latest/contribute/coding_guidelines/index.html?

File: drivers/espi/espi_mchp_xec.c (hunk 1 above)
Comment (albertofloyd) [reply drivers/espi/espi_mchp_xec.c:145]: Commits squashed

**Comment by albertofloyd** [comment]
> Happy to see this change. Can you also update https://docs.zephyrproject.org/latest/contribute/coding_guidelines/index.html?

Updated the coding guidelines too

File: samples/drivers/espi/boards/mec172xmodular_assy6930.conf [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# eSPI + mec172xevb_assy6906
```
Comment (keith-zephyr) [inline samples/drivers/espi/boards/mec172xmodular_assy6930.conf:1]: This new addition to the eSPI samples should be split into a separate commit.

Comment (albertofloyd) [reply samples/drivers/espi/boards/mec172xmodular_assy6930.conf:1]: Moved the config/overlay files into separate commit

**Review by keith-zephyr — DISMISSED** [review]

**Review by jvasanth1 — DISMISSED** [review]

**Comment by aescolar** [comment]
@albertofloyd is this an api change which requires discussion in the arch wg? (CC @carlescufi )

**Review by dleach02 — CHANGES_REQUESTED** [review]
Thank you for doing this. Outside of the question about API change from @aescolar, if this is okay then this PR needs to include the change information in the migration guide (and release notes updated)

**Review by scottwcpg — DISMISSED** [review]

**Comment by ChiHuaL** [comment]
Sorry for the late comment. 
You missed to update the inclusive terms for the file include/zephyr/drivers/espi_saf.h
Also, do you plan to update term `saf` to `taf` and `maf` to `caf`?

**Comment by albertofloyd** [comment]
> Sorry for the late comment. You missed to update the inclusive terms for the file include/zephyr/drivers/espi_saf.h Also, do you plan to update term `saf` to `taf` and `maf` to `caf`?

Yes, plan is to update also eSPI SAF drivers API to move to CAF in separate PR

**Comment by albertofloyd** [comment]
> Thank you for doing this. Outside of the question about API change from @aescolar, if this is okay then this PR needs to include the change information in the migration guide (and release notes updated)

Ack'd

**Comment by albertofloyd** [comment]
> o update also eSPI SAF drivers API to move to CAF in separate PR

> @albertofloyd is this an api change which requires discussion in the arch wg? (CC @carlescufi )

@aescolar @carlescufi  I don't think so, mostly delta between 1.0 and 1.5 are ECN not applicable to Zephyr eSPI driver except the inclusive language.  
 • Merged eSPI Server Addendum Rev0.7 into the eSPI Base specification. Target Attached Flash sharing is applicable for client and server platforms.
• Included ECN- Target Attached Flash RPMC.
• Included ECN- Clarify Flash Erase Length (May 2020)
• Included ECN- BMC integrated RTC (7-11-2020).
• Updated the terms “master/slave” to “controller/target” throughout the specification to align with the use of inclusive language.

**Review by scottwcpg — DISMISSED** [review]

**Review by keith-zephyr — DISMISSED** [review]

**Review by fabiobaltieri — DISMISSED** [review]

File: doc/releases/migration-guide-3.7.rst [hunk 3]
```diff
@@ -120,6 +120,14 @@ Controller Area Network (CAN)
 Display
 =======
 
+Enhanced Seria Peripheral Interface (eSPI)
+==========================================
```
Comment (kartben) [inline doc/releases/migration-guide-3.7.rst:124]:
```suggestion
Enhanced Serial Peripheral Interface (eSPI)
===========================================
```

File: doc/releases/migration-guide-3.7.rst [hunk 4]
```diff
@@ -120,6 +120,14 @@ Controller Area Network (CAN)
...
+  * The macros ESPI_SLAVE_TO_MASTER and ESPI_MASTER_TO_SLAVE were renamed to ESPI_TARGET_TO_CONTROLLER
+    and ESPI_CONTROLLER_TO_TARGET respectively to reflect new terminology in eSPI 1.5 specification.
+  * The enum values ESPI_VWIRE_SIGNAL_SLV_BOOT_STS, ESPI_VWIRE_SIGNAL_SLV_BOOT_DONE and ESPI_VWIRE_SIGNAL_SLV_GPIO_XX
+    were renamed to ESPI_VWIRE_SIGNAL_TARGET_BOOT_STS, ESPI_VWIRE_SIGNAL_TARGET_BOOT_DONE and
+    ESPI_VWIRE_SIGNAL_TARGET_GPIO_XX to reflect new terminology in eSPI 1.5 specification.
```
Comment (kartben) [inline doc/releases/migration-guide-3.7.rst:129]: consider enclosing code litterals between double backquotes.

**Review by kartben — CHANGES_REQUESTED** [review]
btw, are there plans to graduate eSPI to a Stable API?

**Review by scottwcpg — DISMISSED** [review]

File: doc/releases/migration-guide-3.7.rst (hunk 3 above)
Comment (albertofloyd) [reply doc/releases/migration-guide-3.7.rst]: Fixed

File: doc/releases/migration-guide-3.7.rst (hunk 4 above)
Comment (albertofloyd) [reply doc/releases/migration-guide-3.7.rst:129]: Done

**Review by keith-zephyr — DISMISSED** [review]

File: doc/releases/release-notes-3.7.rst [hunk 5]
```diff
@@ -120,6 +120,17 @@ Drivers and Sensors
...
+* eSPI
+
+  * The macros ``ESPI_SLAVE_TO_MASTER`` and ``ESPI_MASTER_TO_SLAVE`` were renamed to
+    ``ESPI_TARGET_TO_CONTROLLER`` and ``ESPI_CONTROLLER_TO_TARGET`` respectively to reflect
+    new terminology in eSPI 1.5 specification.
+  * The following virtual enum values ``ESPI_VWIRE_SIGNAL_SLV_BOOT_STS``,
+    ``ESPI_VWIRE_SIGNAL_SLV_BOOT_DONE`` and ``ESPI_VWIRE_SIGNAL_SLV_GPIO_<NUMBER>`` were renamed to
+    ``ESPI_VWIRE_SIGNAL_TARGET_BOOT_STS``, ``ESPI_VWIRE_SIGNAL_TARGET_BOOT_DONE`` and
+    ``ESPI_VWIRE_SIGNAL_TARGET_GPIO_<NUMBER>`` respectively to reflect new terminology
+    in eSPI 1.5 specification.
```
Comment (kartben) [inline doc/releases/release-notes-3.7.rst:132]: this shouldn't be paraphrasing the migration guide entry. Either drop altogether or, probably better, just make it a simpler bullet point "Renamed several eSPI APIs to adopt terminology from version 1.5 of the specification" or something like this?

**Review by kartben — CHANGES_REQUESTED** [review]
this is really great, thanks for doing this. I have a minor request for change re: release notes entry. Thanks!

File: doc/releases/release-notes-3.7.rst (hunk 5 above)
Comment (albertofloyd) [reply doc/releases/release-notes-3.7.rst]: Summarized

**Comment by albertofloyd** [comment]
@kartben updated release notes

**Review by kartben — APPROVED** [review]

**Review by scottwcpg — APPROVED** [review]

**Review by keith-zephyr — APPROVED** [review]

**Comment by scottwcpg** [comment]
@dleach02 @albertofloyd Hi, can we please get this merged? I have a new SoC eSPI driver and would like to have this done before submitting the new driver. Please, please, please...

**Comment by fabiobaltieri** [comment]
@dleach02 ping, you have a blocker on this

**Comment by albertofloyd** [comment]
ping @dleach02

**Review by dleach02 — APPROVED** [review]

**Comment by kartben** [comment]
@albertofloyd what's the plan with Kconfig `CONFIG_ESPI_SLAVE`, shouldn't it be renamed to `CONFIG_ESPI_TARGET`?
Thanks again for this work!

**Comment by albertofloyd** [comment]
> @albertofloyd what's the plan with Kconfig `CONFIG_ESPI_SLAVE`, shouldn't it be renamed to `CONFIG_ESPI_TARGET`? Thanks again for this work!

Working in this PR https://github.com/zephyrproject-rtos/zephyr/pull/73887
